lua/weapons/weapon_golfclub/shared.lua

```diff
@@ -211,268 +208,258 @@ function SWEP:SetupDataTables()
 end
+
 function SWEP:FinishStroke(hole)
 	local joke = (hole=="Srsly2hard4u")
-local strk = self:GetCurrentStroke()
-if(hole)then
-local endm = strk.." strokes!"
-
-local snd = "hl1/fvox/bell.wav"
-if(strk == 1)then
-endm = "a Hole-in-One!"
-//snd = "skull-trumpet.wav"
-end
-self:GetOwner():EmitSound(snd)
-if joke then
-	local vd = self.Owner:GetPos()
-	timer.Simple(0.4, function() sound.Play("golfmines/amazing.wav", vd, 70, 100, 1) end)
-	self.CantPlaceBall = true
-	timer.Simple(7,function() if IsValid(self) then self.CantPlaceBall=false end end)
-end
-local record = 10
-if SERVER then
-	record = tonumber(self:GetOwner():GetPData("golf_"..hole, 10))
-	if strk<record then
-		self:GetOwner():SetPData("golf_"..hole, tostring(strk))
-	end
-end
-if !joke then GolfNotify("[yellow]"..self:GetOwner():Nick() .." completed hole '"..hole.. "' with "..endm) end
-if SERVER then
-	local iwonsomething = false
-	if strk<record then
-		local strokes = " strokes"
-		if strk==1 then strokes = " stroke" end
-		self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]You beat your record! New record: [orange]"..tostring(strk)..strokes )
-		local t = GolfPrizeTargets[hole]
-		if t then
-			for k,v in pairs(t) do
-				if record > v[1] and strk <= v[1] then
-					if self:GetOwner().familyshared then
-						self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]Sorry, you're not eligible for this prize ;hahaha;" )
-					else
-						self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]Congratulations! You won [gold]"..tostring(v[2]).." points!" )
-						self:GetOwner():PS_GivePoints(v[2])
-					end
-					iwonsomething = true
-				end
-			end
+	local strk = self:GetCurrentStroke()
+	if(hole)then
+		local endm = strk.." strokes!"
+		
+		local snd = "hl1/fvox/bell.wav"
+		if(strk == 1)then
+			endm = "a Hole-in-One!"
+			//snd = "skull-trumpet.wav"
 		end
-	else
-		local strokes = " strokes"
-		if record==1 then strokes = " stroke" end
-		self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]Your record for this hole is: [orange]"..tostring(record)..strokes )
-	end
-	local t = GolfPrizeTargets[hole]
-	if t then
-		local got = math.min(strk,record)
-		local nex = nil
-		for k,v in pairs(t) do
-			if got > v[1] then
-			if nex == nil or nex[1]<v[1] then
-				nex = v
+		self:GetOwner():EmitSound(snd)
+		if joke then
+			local vd = self.Owner:GetPos()
+			timer.Simple(0.4, function() sound.Play("golfmines/amazing.wav", vd, 70, 100, 1) end)
+			self.CantPlaceBall = true
+			timer.Simple(7,function() if IsValid(self) then self.CantPlaceBall=false end end)
+		end
+		local record = 10
+		if SERVER then
+			record = tonumber(self:GetOwner():GetPData("golf_"..hole, 10))
+			local monthlyreset = os.date("*t",tonumber(self:GetOwner():GetPData("monthly_reset",0)))
+			local curdate = os.date("*t",os.time())
+			if curdate["year"] > monthlyreset["year"] or curdate["month"] > monthlyreset["month"] then
+				for h,_ in pairs(GolfPrizeTargets) do
+					self:GetOwner():SetPData("golf_"..h,10)

```

Comment:
Can we keep the all-time best scores and make a seperate monthly best score tracker?